src/Features/Core/Completion/ICompletionService.cs

```diff
@@ -51,6 +46,25 @@ internal interface ICompletionService : ILanguageService
         Task<string> GetSnippetExpansionNoteForCompletionItemAsync(CompletionItem completionItem, Workspace workspace);
 
         /// <summary>
+        /// Called when a completion item is committed.  The completion service can then use this to
+        /// help inform future completion requests.  For example, it can use this information to 
+        /// help decide if this item is 'better' than another when queried.
+        /// </summary>
+        void CompletionItemComitted(CompletionItem item);
+
+        /// <summary>
+        /// Returns true if 'item1' is a better match for the filter text typed so far versus 
+        /// 'item2'.  The better match will be selected after text is typed.
```

Comment:
> The better match will be selected after text is typed.

This may be misleading. I am guessing that the behavior is the "best" match will be selected at the end, which *may* be the better of the two for any particular call to `IsBetterFilterMatch`.

Also, I would recommend considering the following alternative signature, specifically to allow unordered preference and make it clear what the return value means:

```csharp
/// <summary>
/// Determines which of two <see cref="CompletionItem"/> instances is a higher-quality match to the current filter text.
/// </summary>
/// <returns>
/// <para>Greater than 0, if <paramref name="first"/> is a higher quality than <paramref name="second"/>.</para>
/// <para>-or-</para>
/// <para>Less than 0, if <paramref name="second"/> is a higher quality match than <paramref name="first"/>.</para>
/// <para>-or-</para>
/// <para>0, if <paramref name="first"/> and <paramref name="second"/> are equivalent-quality matches.</para>
int CompareMatchQuality(CompletionItem first, CompletionItem second);
```